ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -170,6 +175,31 @@ public int execute() {
     return 0;
   }
 
+  private void initiateAuthorizationDumpTask(Path currentDumpPath) throws SemanticException, IOException {
+    if (RANGER_AUTHORIZER.equalsIgnoreCase(conf.getVar(HiveConf.ConfVars.REPL_AUTHORIZATION_PROVIDER_SERVICE))) {
+      Path rangerDumpRoot = new Path(currentDumpPath, ReplUtils.REPL_RANGER_BASE_DIR);
+      FileSystem fs = rangerDumpRoot.getFileSystem(conf);
+      if (fs.exists(new Path(rangerDumpRoot, ReplAck.RANGER_DUMP_ACKNOWLEDGEMENT.toString()))) {
+        LOG.info("Ranger Authorization Metadata is already exported at {} ", rangerDumpRoot);
+      } else {
+        LOG.info("Exporting Authorization Metadata at {} ", rangerDumpRoot);
+        RangerDumpWork rangerDumpWork = new RangerDumpWork(rangerDumpRoot, work.dbNameOrPattern);
+        Task<RangerDumpWork> rangerDumpTask = TaskFactory.get(rangerDumpWork, conf);
+        if (childTasks == null) {
+          childTasks = new ArrayList<>();
+        }
+        childTasks.add(rangerDumpTask);
```

Comment:
Not getting added to the DAG, how does it get executed?

Reply:
Its getting added before the data copy tasks